packages/engine/crawlers/solana/subscription.ts

```diff
@@ -87,16 +100,18 @@ export const watchLogs = async (
 				});
 			})();
 			const tokenBalances = transaction?.meta?.postTokenBalances?.filter(
-				(balance) => balance.owner === pubkey.toJSON(),
+				(balance) => balance.owner === pubkey.toString(),
 			);
 
 			for (const balance of tokenBalances || []) {
 				if (balance.uiTokenAmount.decimals === 0) {
-					const mpl = new Metaplex(connection);
-					const nft = await mpl
-						.nfts()
-						.findByMint({ mintAddress: new PublicKey(balance.mint) });
-					addCollectibleToState(connection, endpoint, address, nft);
+					if (balance.uiTokenAmount.amount === '1') {
```

Comment:
you can use `if (balance.uiTokenAmount.decimals === 0 && balance.uiTokenAmount.amount === '1')`

Reply:
also consider check it !== '0', for semi-fungible assets?